caluma_interval/client.py

```diff
@@ -1,11 +1,58 @@
 import json
+import logging
+from datetime import datetime, timedelta
 
 import requests
+from envparse import env
+from oauthlib.oauth2 import BackendApplicationClient
+from requests_oauthlib import OAuth2Session
+
+logger = logging.getLogger(__name__)
 
 
 class CalumaClient:
-    def __init__(self, endpoint):
+    def __init__(
+        self,
+        endpoint=env("CALUMA_URI", default="http://caluma/graphql"),
+        oidc_client_id=env("OIDC_CLIENT_ID", default=None),
+        oidc_client_secret=env("OIDC_CLIENT_SECRET", default=None),
+        oidc_token_uri=env("OIDC_TOKEN_URI", default=None),
+    ):
         self.endpoint = endpoint
+        self.oidc_client_id = oidc_client_id
+        self.oidc_client_secret = oidc_client_secret
+        self.oidc_token_uri = oidc_token_uri
+
+        self._token = None
+
+    @property
+    def token(self):
```

Comment:
Style: I kind of dislike `@property` decorators for things that *do* stuff, like communicate with the "outside". Why not make it a regular method?

Reply:
Done!